fcli-core/fcli-fod/src/main/java/com/fortify/cli/fod/scan/cli/cmd/FoDScanStartSastCommand.java

```diff
@@ -13,99 +13,63 @@
 
 package com.fortify.cli.fod.scan.cli.cmd;
 
-import java.io.File;
-import java.util.Properties;
-
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fortify.cli.common.output.transform.IActionCommandResultSupplier;
 import com.fortify.cli.common.output.transform.IRecordTransformer;
 import com.fortify.cli.common.progress.cli.mixin.ProgressWriterFactoryMixin;
-import com.fortify.cli.common.progress.helper.IProgressWriterI18n;
 import com.fortify.cli.common.util.FcliBuildPropertiesHelper;
 import com.fortify.cli.common.util.StringUtils;
 import com.fortify.cli.fod._common.cli.mixin.FoDDelimiterMixin;
 import com.fortify.cli.fod._common.output.cli.AbstractFoDJsonNodeOutputCommand;
 import com.fortify.cli.fod._common.output.mixin.FoDOutputHelperMixins;
-import com.fortify.cli.fod._common.util.FoDEnums;
 import com.fortify.cli.fod.release.cli.mixin.FoDReleaseByQualifiedNameOrIdResolverMixin;
-import com.fortify.cli.fod.scan.cli.mixin.FoDEntitlementPreferenceTypeMixins;
-import com.fortify.cli.fod.scan.cli.mixin.FoDInProgressScanActionTypeMixins;
-import com.fortify.cli.fod.scan.cli.mixin.FoDRemediationScanPreferenceTypeMixins;
-import com.fortify.cli.fod.scan.helper.FoDScanAssessmentTypeDescriptor;
 import com.fortify.cli.fod.scan.helper.FoDScanHelper;
-import com.fortify.cli.fod.scan.helper.FoDScanType;
 import com.fortify.cli.fod.scan.helper.sast.FoDScanSastHelper;
 import com.fortify.cli.fod.scan.helper.sast.FoDScanSastStartRequest;
 import com.fortify.cli.fod.scan_config.helper.FoDScanConfigSastDescriptor;
-
 import kong.unirest.UnirestInstance;
 import lombok.Getter;
-import picocli.CommandLine;
 import picocli.CommandLine.Command;
 import picocli.CommandLine.Mixin;
 import picocli.CommandLine.Option;
 
+import java.io.File;
+import java.util.Properties;
+
 @Command(name = FoDOutputHelperMixins.StartSast.CMD_NAME)
 public class FoDScanStartSastCommand extends AbstractFoDJsonNodeOutputCommand implements IRecordTransformer, IActionCommandResultSupplier {
     @Getter @Mixin private FoDOutputHelperMixins.StartSast outputHelper;
-    
+
     @Mixin private FoDDelimiterMixin delimiterMixin; // Is automatically injected in resolver mixins
     @Mixin private FoDReleaseByQualifiedNameOrIdResolverMixin.PositionalParameter releaseResolver;
-    @Option(names = {"--entitlement-id"})
-    private Integer entitlementId;
-    @Option(names = {"--purchase-entitlement"})
-    private final Boolean purchaseEntitlement = false;
+
+    @Option(names = {"--remediation"})
```

Comment:
I like the (implementation) simplicity of this approach, but shouldn't we still provide users the option to 'start remediation scan if available'? From a user perspective, wouldn't you actually always want to do a remediation scan if available (assuming this has no technical impact, but does reduce the number of entitlements consumed). If so, shouldn't we just remove any such options, and always use a remediation scan if available?

Reply:
OK, I'm not that familiar with FoD, agree to keep this as an option based on your explanation. However, wouldn't it be more user friendly to provide an option 'use remediation scan if available', rather than plain remediation=true|false?

For single scans, if no remediation scan is available, I'd assume that `--remediation=true` would fail, correct? So, to avoid pipelines from failing, while at the same time minimizing entitlement consumption, for single scans it would make sense for users to specify something like `--remediation=IfAvailable`, wouldn't it?

For subscription scans, I guess `--remediation=IfAvailable` would act the same as `--remediation=No`.